### PR TITLE
#2948 Fixed bug

### PR DESCRIPTION
**Pull request by nazneennahar**
## The basics

- [ x] I branched from develop
- [ ] My pull request is against develop
- [ ] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Resolves

#2948
### Proposed Changes

The pull request resolves the issue of breaking the entire block when a new block is being inserted.
### Reason for Changes

These changes helps the game to function properly
### Test Coverage

Tested on:
yes

### Documentation

NO

### CODE REVIEW

File: core/insertion_marker_manager.js [hunk 1]
```diff
@@ -406,7 +406,10 @@ Blockly.InsertionMarkerManager.prototype.shouldReplace_ = function() {
   var local = this.localConnection_;
 
   // Dragging a block over an existing block in an input.
-  if (local.type == Blockly.OUTPUT_VALUE) {
+  if (
```
Comment (rachel-fenichel) [inline core/insertion_marker_manager.js]: Reformat:
```
if (local.type == Blockly.OUTPUT_VALUE ||
    local.type == Blockly.PREVIOUS_STATEMENT) {
```

**Comment by BeksOmega** [comment]
Turns out this actually breaks the zelos renderer.

Steps to reproduce:
1. Pull down the example blocks I put up in #3525 
2. Import the following xml:
```
<xml xmlns="https://developers.google.com/blockly/xml">
  <block type="test_connections_stack_next">
    <next>
      <block type="test_connections_stack_blue">
        <next>
          <block type="test_connections_stack_yellow">
            <next>
              <block type="test_connections_stack_yellow">
                <next>
                  <block type="test_connections_stack_red">
                    <next>
                      <block type="test_connections_stack_prev"/>
                    </next>
                  </block>
                </next>
              </block>
            </next>
          </block>
        </next>
      </block>
    </next>
  </block>
</xml>
```
3. Drag the red block down to disconnect.
4. View the error in console.

This is caused by the `highlightBlock_` function duplicating the local.type check changed by this PR. It could be fixed by simply adding the `|| PREVIOUS_STATEMENT` check inside that function as well, but that seems pretty smelly to me (not very DRY).

The `highlightBlock_` function is handling logic it shouldn't (the description says it should handle the visual for replacing blocks, but in zelos it is handling insertion as well), so I am going to attempt to fix it at that level.

If I am unable to fix it in a timely manner, I will write up a more detailed description of the problem.

**Comment by BeksOmega** [comment]
This PR also breaks insertion for typed statements:
TypedInsertion

Observe how it displays a "replace" preview, but the block actually gets inserted.

This is because the `Connection.lastConnectionInRow` function doesn't support stacks.

**Comment by rachel-fenichel** [comment]
Closing as stale.

**Pull request closed without merging**